### PR TITLE
Lots Todo and dependent things

### PR DESCRIPTION
**Pull request by nelix**
screen shot 2015-02-26 at 2 45 12 pm

* general refactoring
* some bugs
* single line edits in TODOS (and thus cancelable-edit)
* danger alerts when deleting todo items
* use Link component in cancelable-edit's creating mode
* make delete button centre properly
* use inlay component in delete dialog

### CODE REVIEW

File: packages/delete-button/index.jsx [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
 
+require('./style.scss');
+
 module.exports = React.createClass({
   displayName: 'DeleteButton',
 
   render() {
     return (
-      <button {...this.props} className={'DeleteButton' + (this.props.className ? ` ${this.props.className}` : '')}>
+      <button {...this.props} className={this.constructor.displayName + (this.props.className ? ` ${this.props.className}` : '')}>
```
Comment (grapegravity) [inline packages/delete-button/index.jsx]: I don’t mind this, but I’d like to hear input from others on whether they’re okay with it.

File: packages/todo/todo-item.jsx [hunk 2]
```diff
@@ -53,22 +55,24 @@ module.exports = React.createClass({
   },
 
   render() {
-    const className = React.addons.classSet({
-      'Todo-listItem--complete': this.props.completed,
-      'Todo-listItem-edit': true,
+    const className = new SuitClassSet('TodoItem').
+      createDescendent('edit');
```
Comment (grapegravity) [inline packages/todo/todo-item.jsx]: This is an interesting use of that API… I don’t think it’s too worthwhile unless you’re using the `TodoItem` instance, too?

File: packages/todo/todo-item.jsx [hunk 3]
```diff
@@ -88,11 +92,12 @@ module.exports = React.createClass({
               onOk={this.handleConfirmOk}
               onCancel={this.handleConfirmCancel}
               okText="Delete"
+              danger
             >
-            Are you sure you want to delete: <pre style={{display: 'inline', borderRadius: 3, background: '#eee', padding: 3}}>
+            Are you sure you want to delete: <Inlay>
                 {this.props.name.substr(1, MAX_ALERT_LENGTH)}
```
Comment (grapegravity) [inline packages/todo/todo-item.jsx]: Shouldn’t this start at `0`?

File: packages/todo/todo-item.jsx (hunk 2 above)
Comment (nelix) [inline packages/todo/todo-item.jsx]: pretty good point, I think I had some other features which I removed... I'll fix it up!

Comment (nelix) [inline packages/todo/todo-item.jsx]: What should I do about 
```
    className.addModifier({
      'complete': this.props.completed,
    });
```

Comment (grapegravity) [inline packages/todo/todo-item.jsx]:
```
const complete = this.props.completed;
className.addModifier({
  complete,
});
```
maybe?

Comment (nelix) [inline packages/todo/todo-item.jsx]: :open_mouth:  es6 is so cool

**Pull request merged**